From 5403541355fe9d511591a10850d4e141f038a30b Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Tue, 5 Oct 2021 21:01:15 -0700
Subject: [PATCH 11/21] SystemUI: Always refresh power menu on UI mode change

This is necessary for reliably theming the global actions dialog with
dynamic colors and adapting it to light/dark themes.

Demo screenshots (with color overlays applied):
https://twitter.com/kdrag0n/status/1445960685427433473

Change-Id: Id6f6623096ed231e841ca3794b47f8a0ab54dae6
---
 .../android/systemui/globalactions/GlobalActionsDialogLite.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
index 65c520ef2ff4..43a1084540f6 100644
--- a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
+++ b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
@@ -762,7 +762,7 @@ public class GlobalActionsDialogLite implements DialogInterface.OnDismissListene
     @Override
     public void onUiModeChanged() {
         mContext.getTheme().applyStyle(mContext.getThemeResId(), true);
-        if (mDialog != null && mDialog.isShowing()) {
+        if (mDialog != null) {
             mDialog.refreshDialog();
         }
     }
-- 
2.25.1

